pkg/action/scan.go

```diff
@@ -36,8 +36,11 @@ var (
 	// compiledRuleCache are a cache of previously compiled rules.
 	compiledRuleCache atomic.Pointer[yarax.Rules]
 	// compileOnce ensures that we compile rules only once even across threads.
-	compileOnce         sync.Once
-	ErrMatchedCondition = errors.New("matched exit criteria")
+	compileOnce           sync.Once
+	ErrMatchedCondition   = errors.New("matched exit criteria")
+	ErrFileScan           = NewFileReportError(nil, "", "scan failed")
+	ErrGenerateFileReport = NewFileReportError(nil, "", "failed to generate file report")
+	ErrNilFileReport      = NewFileReportError(nil, "", "file report is nil")
```

Comment:
Is this an actual error? 🤔 Mostly asking because we have this later:

```
case TypeNilError:
		return nil, nil
```

Reply:
It's technically not an error. We can actually rely on the existing `fr == nil` check to return `nil, nil`.

Reply:
👍 sgtm. document this behavior in the method's godoc! 🙏

Reply:
Updated in [`8e1e5c6` (#777)](https://github.com/chainguard-dev/malcontent/pull/777/commits/8e1e5c6cf378a547f46bccdd7d73ceb9a912eb5b).